### PR TITLE
Bin remove paste

### PR DESCRIPTION
**Pull request by Shir0kamii**
I did a little script you can use to delete pastes from a shell. It's not only a temporary solution waiting for the admin page, but also a way to delete multiple pastes in a single command over ssh.

### CODE REVIEW

File: zerobin/remove_paste.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    """Take either the ID or the URL of a paste, and return its ID"""
+
+    try_url = paste_url.search(paste)
+
+    if try_url:
+        return try_url.group('paste_id')
+    return paste
+
+
+def remove_paste(*pastes, quiet:'q'=False):
```
Comment (Natim) [inline zerobin/remove_paste.py]: Shouldn't it be ``quiet=False``?

**Comment by sametmax** [comment]
Hi,

Nice, thank you. But I need two more things to accept this patch:

- clean argument parsing. You can use clize for that.
- a parameter for zerobin.py to not run the server and delete the post instead. This way, even when the package is not installed using setup.py install or pip, you still can use the script.

:)

File: zerobin/remove_paste.py (hunk 1 above)
Comment (Shir0kamii) [inline zerobin/remove_paste.py]: The annotation 'q' tell clize that the short version of --quiet is -q

**Comment by sametmax** [comment]
Arf, I just noted : zerobin should be able to run on Python 2 and 3 at the same time. I notice this run only on Python 3 :) Can you fix that ? E.G: removing annotations, checking encoding, etc.

**Comment by Shir0kamii** [comment]
I added the compatibility for python 2, but I don't get what you want with the script.

* I already use clize for argument parsing, so what do I have to do ?
* the script for zerobin.py is located in zerobin/routes.py. Do you want me to move the main fonction from routes.py to zerobin.py and add an alternative call `--delete` ?

**Comment by sametmax** [comment]
> I already use clize for argument parsing, so what do I have to do ?

Sorry, missed that.

> the script for zerobin.py is located in zerobin/routes.py. Do you want me to move the main fonction from routes.py to zerobin.py and add an alternative call --delete 

Move it to a cmd.py file, add a --delete-paste (not the cleanest way but clize don't accept subcommands if I recall so i'll have to do), then change the import in zerobin.py

Thanks for spending so much time on this :)

**Comment by Shir0kamii** [comment]
> Sorry, missed that.

No problem

>  add a --delete-paste (not the cleanest way but clize don't accept subcommands if I recall so i'll have to do)

Clize does support subcommands. We can either have `zerobin runserver` and `zerobin delete-paste` or `zerobin` and `zerobin --delete-patste`. Which one do you prefer ?

> Thanks for spending so much time on this :)

It's not that much. It allows me to acquire experience in Python (a language I love) on a real project. So thank you too !

**Comment by sametmax** [comment]
Subcommandes are much better, but if called without a subcommand, can you make it default to runserver ? I don't want to break compat.

Plus, we'll need some doc on this anyway.

**Comment by Shir0kamii** [comment]
I searched in the doc and tried to hack, but didn't find any way to use default subcommand. Then we must use an alt command not to break compat.

**Comment by sametmax** [comment]
There is a way. A bit hacky but it should work: check sys.argv. If it contains one of the subcommand as first argument, use it, otherwise, use the default.

**Comment by Shir0kamii** [comment]
Oh, I didn't think about it ! I'll try to implement it soon.

**Comment by Shir0kamii** [comment]
It should be okay now !

**Comment by sametmax** [comment]
Ty man. I'll add some doc and add your name to the contrib file :)

**Comment by Shir0kamii** [comment]
You're welcome. The admin page should follow soon, but will allow only one deletion at a time.

**Comment by sametmax** [comment]
We don't need more. We have only a few take down notices a year.

**Comment by Shir0kamii** [comment]
It's almost done, so I'll PR anyway. I just have a few bugs to fix, the interface is already done.

PS: I broke zerobin.py, can you fix it ? It's just a `s/routes/cmd/`.

**Comment by sametmax** [comment]
I'll fixe that tomorrow. I did not publish it on pypi yet so it's safe.

**Pull request merged**